### ISSUE TITLE
Change default web behaviors that don't make sense in the context of Electron

### ISSUE THREAD

**Issue by nornagon** [electron/electron#12631]
There are a number of behaviors that Electron inherits from Chromium which don't make sense for the vast majority of Electron apps. For instance:

- dragging a file into a window navigates to that file
- pinch-zooming in a window zooms the UI
- <kbd>Cmd+Click</kbd> opens links a new BrowserWindow

Electron apps that don't want these behaviors currently have to disable them explicitly, in an ad-hoc and error-prone way. Since these behaviors are undesirable for almost every Electron app, we should disable them by default.

This issue is a tracking issue for this general idea; separate issues will be created for specific changes that need to be made.

**Comment by timfish** [comment]
`setVisualZoomLevelLimits` disables pinch zooming. Most production Electron apps I've checked already do this.
```
webContents.setVisualZoomLevelLimits(1, 1);
```
`setZoomFactor` and `setLayoutZoomLevelLimits` disable zoom scaling which should probably be left enabled for accessibility reasons.

**Comment by timfish** [comment]
How about `BrowserWindow` option `acceptFirstMouse`:

> `acceptFirstMouse` Boolean (optional) - Whether the web view accepts a single mouse-down event that simultaneously activates the window. Default is false.

I think the default should be true.

**Comment by nornagon** [comment]
Hmm, I'm not sure I agree—on macOS at least, clicking on most native apps when they're not focused brings the window to the front without causing a mouse-click on whatever it was you clicked on. Clicking a button in a background window requires two clicks. (Perhaps it's different on other OSes though? We could change the default to be OS-specific.)

**Comment by timfish** [comment]
ah yes you're right and it looks like `acceptFirstMouse` is a macOS only option. 

On Windows, Electron already accepts first mouse by default. 👍

**Comment by nornagon** [comment]
Images (and perhaps other elements, like <a> tags?) should be non-draggable by default.

**Comment by nornagon** [comment]
Possibly, text should be non-selectable by default? i.e. `user-select: none`.

**Comment by remhume** [comment]
I was digging through google and issues to find out how to enable pinch-zoom (and only got tons of results on how to disable it, suggesting it's enabled by default) and finally found this thread and #12679, it would be nice to have this documented somewhere more visible.
(At least i couldn't find anything yet)

**Comment by nornagon** [comment]
I agree, but I'm not sure where to add that documentation! Where were you expecting to find it? That's probably the best place to put it :)

**Comment by remhume** [comment]
Hm, i mostly looked at
https://electronjs.org/docs/api/web-frame#webframesetvisualzoomlevellimitsminimumlevel-maximumlevel (this is where the defaults should be listed i think)

and https://electronjs.org/docs/api/web-contents#contentssetvisualzoomlevellimitsminimumlevel-maximumlevel

Is the latter still supported? so far i only had success using webFrame. If not, a reference to webFrame would be helpful too. Until I researched on this had no idea what or why the 'web-frame' renderer export exists, because there was no reference to it in webContents nor BrowserWindow docs (or anywhere else it seems. though the search is not that reliable). Even now i have no clue why we have webFrame and webContents and what the difference is.

**Comment by paulcbetts** [comment]
I am extremely Into all of these things 💖

**Comment by gdavidkov** [comment]
Is still zoom is per-origin?

**Comment by naruaway** [comment]
In terms of the default behavior, shouldn't we provide different presets like "browser-like" (e.g. use electron as web site wrapper or use electron as hackable web browser) or "app-like" (e.g. only show bundled HTML/JS to build native app like UX)? Then using the former preset can keep the existing behavior like "`Cmd+Click` opens links a new BrowserWindow".

I think just changing the default behavior would make it difficult/error-prone to create browser-like experience instead. Also it would be nice to keep Chrome's behavior where possible especially for newcomers with mental models like "electron is just a web browser with hackable API"

In addition to that, [the official electron doc on sandbox](https://www.electronjs.org/docs/latest/tutorial/sandbox#a-note-on-rendering-untrusted-content) is mentioning
> Rendering untrusted content in Electron is still somewhat uncharted territory, though **some apps are finding success (e.g. Beaker Browser).**

So I would assume we should not completely ignore "using electron as hackable web browser" use case, which is currently I am personally currently experimenting with.

**Comment by grantcv1** [comment]
Seven years on and I'm trying to make a tutorial video on the mac for our app and this useless behavior keeps ruining my takes. There is no way out of it so I have to scrap the take, reset my app, and start again -- hoping it doesn't happen again. It's a totally useless feature from which there is no escape other than to restart the application.
